pkg/utils/parsearguments.go

```diff
@@ -0,0 +1,75 @@
+package utils
+
+import (
+	"fmt"
+	"strings"
+)
+
+// StringMap implements the flag.Value interface and allows to parse stringified maps
+// in the form: "key1=val1,key2=val2".
+type StringMap map[string]string
+
+// String returns the stringified map.
+func (sm *StringMap) String() string {
+	return fmt.Sprintf("%#v", *sm)
```

Comment:
AFAIU, the String method should return a representation as expected by the Set() function, as it may be used to represent default values. Additionally, it should work also with a [nil receiver](https://pkg.go.dev/flag#Value).